common/persistence/cassandra/metadata_store.go

```diff
@@ -126,13 +126,17 @@ func (m *MetadataStore) CreateNamespace(
 	request *p.InternalCreateNamespaceRequest,
 ) (*p.CreateNamespaceResponse, error) {
 	query := m.session.Query(templateCreateNamespaceQuery, request.ID, request.Name).WithContext(ctx)
-	applied, err := query.MapScanCAS(make(map[string]interface{}))
+	_, err := query.MapScanCAS(make(map[string]interface{}))
 	if err != nil {
 		return nil, serviceerror.NewUnavailable(fmt.Sprintf("CreateNamespace operation failed. Inserting into namespaces table. Error: %v", err))
 	}
-	if !applied {
-		return nil, serviceerror.NewNamespaceAlreadyExists("CreateNamespace operation failed because of uuid collision.")
-	}
+
+	// let the code falls through and call the next function if there is a uuid collision here (MapScanCas returns false).
```

Comment:
Good point, let me take another look and reconsider.

Reply:
MapScanCAS will populate the passed-in map with the conflicting row I think. and you can check if the name matches.

Reply:
I ran some tests, removed the entry from 'namespaces' database and checked this code path.  We only reach here if `registerNamespace` fails to find `name` in `namespaces` database ([here](https://github.com/temporalio/temporal/blob/0bdbbfc737d469b828a861836c42c1c6650cbfcb/service/frontend/namespace_handler.go#L172)). This is the only path I found that calls this function. So if we are here, it means that `name` does not exist in the `namespaces` database, please let me know if there is any other code path that I might be missing. 

Here either the query returns empty and `applied` is `true`, which means it could not find either `name` or `id` in `namespaces_by_id` table, which means there is no entry in any of these tables. 

If my understanding is correct,  if `applied` is `false` and populates the map, there are two cases: 
1- only `name` matched: which is the code path that leads to this issue. We want to fall through and let it be added to the second table. 
2- both `name` and `id` match, this is similar to the first case, as long as `name` matches, it means it does not exist in the second table. 
3-only `Id` matched: this is duplicate `Id` and most probably the effort to add this to the second table fails as well. But we can probably return an error here.

So to me it looks like, the best action to take here is to return an error about duplicate `ID` for case 3 and for the rest let it fall through. WDYT?

Reply:
I don't think case 1 exists. The first query runs on the `namespace_by_id` table whose primary key is only `id`. So if only name matched -> id not match -> primary key doesn't exist -> query will apply.